creating_images/guidelines.adoc

```diff
@@ -299,44 +299,27 @@ privileged user.
 Because the user ID of the container is generated dynamically, it will not have
 an associated entry in *_/etc/passwd_*. This can cause problems for applications
 that expect to be able to look up their user ID. One way to address this problem
-is to use link:https://cwrap.org/nss_wrapper.html[nss wrapper] and dynamically
-create a *_passwd_* file with the container's user ID as part of the image's
-start script:
+is to dynamically create a *_passwd_* file with the container's user ID as part
+of the image's `ENTRYPOINT` script. This what a Dockerfile might include:
 
 ----
-export USER_ID=$(id -u)
-export GROUP_ID=$(id -g)
-envsubst < ${HOME}/passwd.template > /tmp/passwd
-export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
-export NSS_WRAPPER_PASSWD=/tmp/passwd
-export NSS_WRAPPER_GROUP=/etc/group
+ENV USER_NAME=nobody \
+    USER_UID=99
+RUN chown -R ${USER_UID}:0 /some/directory \
+    && chmod -R g=u /some/directory /etc/passwd
```

Comment:
I would assume those files would be untouched, so they'd still be owned+accessible by the service user.  Hence my original point about not changing the user-ownership of the files.

There are basically two cases to cover:
1) you're running the process as the expected user:  things will work because presumably those files were installed and setup to be accessed by that user

2) you're running things as a random user (member of root group): things will work because you've chgrp'd all the files into the root group and given the root group the necessary permissions on the files.

The problematic case would be:

3) the service account is a member of group foo.  The files were previously owned by user foo and group bar, but now they are owned by root. the service can no longer access the files because the service account is not foo and is not a member of the root group.

however again, what you're doing won't solve that either.  only granting "other/world" permissions would.